Update contrib guides and templates (#21)
eiffel-community/.github: CONTRIBUTING.md

While we welcome requests for changes (in the form of Issues), we absolutely love ready solutions (in the form of Pull Requests). The best requests are the ones with Pull Requests to go along with them.

Contributions can be made by anyone using the standard [GitHub Fork and Pull model](https://help.github.com/articles/about-pull-requests). When making a pull request, keep a few things in mind.
1. Explicitly connect a pull request to an Issue (see exemptions below).
1. Make sure you target the correct branch. If you are unsure which branch is appropriate, ask in the Issue thread.
1. Pull Requests will be publicly reviewed, criticized, and potentially rejected. Don't take it personally.

### Exemptions to Connecting Pull Requests to Issues
There are some circumstances when there makes very limited sense to connect an Issue to a Pull Requests. There is no need to create and connect Issues under the following circumstances:

> - The PR is created towards a new repository with only one active contributor/maintainer.

Reviewer: I'm not sure the number of contributors is the key point. For me it's whether the repo is under active/early development, i.e. there's no point in writing issues while you're in the "getting it to work" state.

Author: Hmm, ok, maybe so. My thinking was that if a full team of people are working on a new project it could still make sense to create PRs for the updates, but maybe that's not always the case. I'll make an attempt to reformulate it.

> - The PR is for a bug/change that is very trivial and thus doesn't expect any deep discussions needed before approving it.

Reviewer: The "expect" part sounds weird to me. How about this?
```suggestion
- The PR is for a bug/change that's very trivial and thus isn't expected to raise any significant discussions before approval.
```
